### ISSUE TITLE
UriTemplate can't recognize variable names that contain a dot

### ISSUE THREAD

**Issue by mxx-td** [spring-projects/spring-hateoas#1800]
to reproduce:

```java
//  work
UriTemplate.of("http://localhost:8201/{?integrations}").getVariableNames().size() // => 1 

// not work
UriTemplate.of("http://localhost:8201/{?integrations.not}").getVariableNames().size() // => 0
```

As a comparison, the ruby client [sporkmonger/addressable ](https://github.com/sporkmonger/addressable) parses variables correctly
```ruby
require 'addressable'
Addressable::Template.new("http://localhost:8201/{?integrations.not}").variables # => ["integrations.not"]
```

below is a screenshot taken from the RFC(https://datatracker.ietf.org/doc/html/rfc6570#section-2.3):
![image](https://user-images.githubusercontent.com/99254663/174271715-8ee389aa-41be-482c-aa45-93cceb0e1961.png)

**Comment by odrotbohm** [comment]
Fixed and backported to 1.5 and 1.4.

**Comment by mxx-td** [comment]
that's fast. thank you
